### DISCUSSION TITLE
Insertion duplication

### DISCUSSION THREAD

**Discussion by Walterhart** [supabase/supabase#10599] [Questions] [answered]
I made a  insertion function for a table base on given  argument. My query works but I was wondering is there a way to not insert duplications  in table?
Note: I have tried primary keys (id,video_id, user_id).
  
  ```
create or replace function add_favorite_hbo(user_num uuid, video_num text)
  returns void
  language sql
  as $$
  insert into hbo_favorite (user_id, video_id)
  values(user_num,video_num)
  $$;

```
my table below:
![image](https://user-images.githubusercontent.com/70658679/204439007-0ec790ff-d6a3-42e4-a410-8b9e8fb054ee.png)

**Comment by GaryAustin1** [comment, marked as answer]
By selecting multiple columns for primary key that means only the combination of all 3 is considering unique.  
If you violate a unique constraint with insert you will error.   You can do an "upsert" operation by adding `on conflict (name) do update` clause to the insert operation... https://www.postgresql.org/docs/current/sql-insert.html
